.github/workflows/web-deploy-ipfs.yml

```diff
@@ -0,0 +1,56 @@
+name: Web Deploy to IPFS
+
+on:
+  pull_request:
+    branches:
+      - dev
+    paths:
+      - apps/web/**
+      - packages/**
+      - .github/workflows/web-deploy-ipfs.yml
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read       
+      statuses: write      
+      checks: write        
+      pull-requests: write 
+
+    name: Deploy to IPFS
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: ./.github/actions/yarn
+
+      - uses: ./.github/actions/build
+        with:
+          secrets: ${{ toJSON(secrets) }}
+          prod: ${{ true }}
+
+      - name: Remove .html extension from all files
```

Comment:
Do you remove the `.html` extension so that next page routes are loaded correctly via IPFS Gateways? 

If so, you can fix this more elegantly by setting [`trailingSlash` to `true` in your `next.config.js` file](https://nextjs.org/docs/app/api-reference/config/next-config-js/trailingSlash). 

The reason is that IPFS Gateways expect either a filename with an extension, and if none is provided, they will look for the `index.html`. 

Setting [`trailingSlash` to `true` in your `next.config.js` file](https://nextjs.org/docs/app/api-reference/config/next-config-js/trailingSlash) will create a folder for each next route with an `index.html` making it play nicely with gateways.

Reply:
As far as I know, this is something we cannot use which would break the traditional hosting app. Maybe @katspaugh you can explain a bit more (this was already discussed in this [comment](https://github.com/safe-global/safe-wallet-monorepo/pull/5165#discussion_r1975585903))